src/public-assets/js/dependent-dropdown.js

```diff
@@ -0,0 +1,21 @@
+var Dropdown = (function ($) {
+
+    return {
+        init: function (formEl) {
+            $(formEl).on('change', 'select.dep-drop',  function () {
+                var $element = $(this);
+                $element.parent().next('.subs').remove();
```

Comment:
A few more comments here:

1. This part makes an assumption about the HTML structure (`$().parent().next()`...). I think a better way would be to use a data- attribute, with its value set to the id/selector of the next element. Something like `data-target='#element'` should do.
2. I noticed here that the next element (or its parent) is removed and then rebuilt. I want to suggest a different approach:
- You remove all the `<option>` elements in the next select. 
- You disable the next select element and add the loading class.
- You only return the new `<option>` tags from the API request and add it to the next select element.
- You enable the next select element and remove the loading class.

Reply:
@olajideoye the number of dependent/child dropdowns is unknown beforehand, therefore all the children dropdowns are detached on change and replaced entirely because of the multi-level dependency. Please see recent updates. Thank you!